src/Octoshift/ReclaimService.cs

```diff
@@ -0,0 +1,246 @@
+            /// Checks if the user has been claimed at least once (regardless of the last reclaiming result)
+            /// </summary>
+            /// <param name="login"></param>
+            /// <param name="id"></param>
+            /// <returns></returns>
+            public bool IsClaimed(string login, string id)
+            {
+                return _mannequins.FirstOrDefault(m =>
+                    login.Equals(m.Login, StringComparison.OrdinalIgnoreCase) &&
+                    id.Equals(m.Id, StringComparison.OrdinalIgnoreCase)
+                    && m.MappedUser != null)?.Login != null;
+            }
+
+            public bool IsClaimed(string login)
+            {
+                return _mannequins.FirstOrDefault(m =>
+                    login.Equals(m.Login, StringComparison.OrdinalIgnoreCase)
+                    && m.MappedUser != null)?.Login != null;
+            }
+
+            public bool IsEmpty()
+            {
+                return _mannequins.Length == 0;
+            }
+
+            public IEnumerable<Mannequin> GetUniqueUsers()
+            {
+                return _mannequins.DistinctBy(x => $"{x.Id}__{x.Login}");
+            }
+        }
+
+        public ReclaimService(GithubApi githubApi, OctoLogger logger)
+        {
+            _githubApi = githubApi;
+            _log = logger;
+        }
+
+        public virtual async Task ReclaimMannequin(string mannequinUser, string mannequinId, string targetUser, string githubOrg, bool force)
+        {
+            var githubOrgId = await _githubApi.GetOrganizationId(githubOrg);
+
+            var mannequins = new Mannequins((await GetMannequins(githubOrgId)).GetByLogin(mannequinUser, mannequinId));
+            if (mannequins.IsEmpty())
+            {
+                throw new OctoshiftCliException($"User {mannequinUser} is not a mannequin.");
+            }
+
+            // (Potentially) Save one call to the API to get the targer user
+            // // (it also makes it unnecessary to check for claimed users during reclaiming loop)
+            if (!force && mannequins.IsClaimed(mannequinUser))
+            {
+                throw new OctoshiftCliException($"User {mannequinUser} is already mapped to a user. Use the force option if you want to reclaim the mannequin again.");
+            }
+
+            var targetUserId = await _githubApi.GetUserId(targetUser);
+            if (targetUserId == null)
+            {
+                throw new OctoshiftCliException($"Target user {targetUser} not found.");
+            }
+
+            var failed = false;
+
+            // get all unique mannequins by login and id and map them all to the same target
+            foreach (var mannequin in mannequins.GetUniqueUsers())
+            {
+                var result = await _githubApi.ReclaimMannequin(githubOrgId, mannequin.Id, targetUserId);
+
+                failed |= HandleResult(mannequinUser, targetUser, mannequin, targetUserId, result);
```

Comment:
This is not going to be a blocking comment so I'll leave it up to you, but it took me a while to figure out what happens here. It's weird if `HandleResult()` returns `true` it actually means failure! So I would inverse the logic and instead of `failed` would create `success` flag and then if `HandleResult` returned `true` it would mean `success` instead of `failure`. Something like this:
```c#
var success = true;
foreach (...)
{
    ...
    success &= HandleResult(...);
}

if (!success)
{
    throw ...
}
```

Reply:
changed